examples/two_moons_classification_ood.pct.py

```diff
@@ -126,12 +141,83 @@ def plot_uncertainty(
 )
 
 # %%
-plot_uncertainty(prob_model, test_data_loader, grid_size=100)
+test_modes = compute_test_modes(prob_model, test_data_loader)
+grid, grid_inputs_loader = get_grid_inputs_loader(grid_size=100)
+grid_entropies = prob_model.predictive.entropy(grid_inputs_loader)
+plot_uncertainty_over_grid(grid=grid, scores=grid_entropies, test_modes=test_modes)
+plt.show()
+
+# %% [markdown]
+# Clearly, the model is overconfident on inputs that are far away from the training data.
+# This behaviour is not what one would expect, as we rather the model being less confident on out-of-distributin inputs.
+
+# %% [markdown]
+# ### Fit an OOD classifier to distinguish between in-distribution and out-of-distribution inputs
+# Given the trained model from above, we can now use one of the models provided by Fortuna to actually improve
+# the model's confidence on the out-of-distribution inputs.
+# In the example below we will use the Malahanobis-based classifier introduced in
+# [Lee, Kimin, et al](https://proceedings.neurips.cc/paper/2018/file/abdeb6f575ac5c6676b747bca8d09cc2-Paper.pdf)
+
+# %%
+from fortuna.ood_detection.mahalanobis import MalahanobisClassifierABC
+from fortuna.model.mlp import DeepResidualFeatureExtractorSubNet
+from functools import partial
+import jax
+
+
+class MalahanobisClassifier(MalahanobisClassifierABC):
+    @partial(jax.jit, static_argnums=(0,))
+    def apply(self, inputs, params, mutable, **kwargs):
```

Comment:
This unfortunately I don't like as not enough user friendly. We shouldn't ask users  to 1/ remember `feature_extractor_subnet` is in `self`, 2/ know the structure of `params` and 3/ update mutables themselves.

In principle, I don't see why the `apply` function needs to be given by the user. Similarly to `ProbModel`, given a feature extractor model, we can have an internal model manager that handles the forward pass. 

Doing it this way, there is also no need for the user to inherit a base class. The user should simply pass a (feature extractor) model to the OOD method in use. We should handle the rest.